packages/truffle-interface-adapter/lib/web3-shim.js

```diff
@@ -3,30 +3,50 @@ const BN = require("bn.js");
 
 // This is a temporary shim to support the basic issues with Quorum
 
-class Web3Shim {
+class Web3Shim extends Web3 {
```

Comment:
Favor composition over inheritance, no? Why extend Web3?

Reply:
This is a Web3 shim. It's purpose is to shim around web3. So for consumers of it, it should look "more or less" like web3. This is all "temporary" until a better system is architected.

The inheritance was driven by stuff I'm doing for `axcore` integration, but I recently discovered that the benefits of inheritance don't translate with `Web3`'s implementation. It's not worth changing it back now.

Reply:
Wait so we're planning on replacing this entirely? I thought we were going to continue morphing it to meet our needs.

Reply:
Much of this code will be reused but the morphing will look like something entirety different after a better architecture is designed. The whole point of doing this PR now instead of 1-2 months from now was to get something out the door.

Reply:
Further discussion happened over Zoom, and I added a comment (https://github.com/trufflesuite/truffle/pull/1806/commits/4457aa67a4692645ab34f389442a96bd21d684d5) to better explain the nature of the Web3Shim in the event that it becomes less temporary than planned.